src/Symfony/Component/Process/ProcessManager.php

```diff
@@ -0,0 +1,178 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Process;
+
+use SplQueue;
+use Symfony\Component\Process\Exception\LogicException;
+use Symfony\Component\Process\Exception\ProcessFailedException;
+use Symfony\Component\Process\Exception\RuntimeException;
+
+/**
+ * Manages running processes in parallel.
+ *
+ * @author John Nickell <email@johnnickell.com>
+ */
+class ProcessManager
```

Comment:
I would like to see a different name here, what about something towards `ParallelProcessRunner` or `ProcessRunnerQueue`? [Naming Classes - How to avoid calling everything a "<WhatEver>Manager"?](https://stackoverflow.com/questions/1866794/naming-classes-how-to-avoid-calling-everything-a-whatevermanager)

Also from an architectural point of view, I'd like to see an interface implemented here and the constants either in that interface or a dedicated final (with private constructor) class. This will avoid difference references and gives a proper decoration possibility.